packages/flutter/test/material/tabs_test.dart

```diff
@@ -1754,4 +1771,54 @@ void main() {
 
   });
 
+  testWidgets('Skipping tabs with a KeepAlive child works', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/11895
+    final List<String> tabs = <String>[
+      'Tab1',
+      'Tab2',
+      'Tab3',
+      'Tab4',
+      'Tab5',
+    ];
+    final TabController controller = new TabController(
+      vsync: const TestVSync(),
+      length: tabs.length,
+    );
+    await tester.pumpWidget(
+      new MaterialApp(
+        home: new Align(
+          alignment: Alignment.topLeft,
+          child: new SizedBox(
+            width: 300.0,
+            height: 200.0,
+            child: new Scaffold(
+              appBar: new AppBar(
+                title: const Text('tabs'),
+                bottom: new TabBar(
+                  controller: controller,
+                  tabs: tabs.map((String tab) => new Tab(text: tab)).toList(),
+                ),
+              ),
+              body: new TabBarView(
+                controller: controller,
+                children: <Widget>[
+                  new AlwaysKeepAliveWidget(),
+                  const Text('2'),
+                  const Text('3'),
+                  const Text('4'),
+                  const Text('5'),
+                ],
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+    expect(find.text(AlwaysKeepAliveWidget.text), findsOneWidget);
+    await tester.tap(find.text('Tab4'));
+    await tester.pumpAndSettle();
+    await tester.pump();
+    expect(controller.index, 3);
+    expect(find.text('4'), findsOneWidget);
```

Comment:
maybe assert at the beginning that text('4') didn't find anything to ensure that the setup is and stays correct? And assert at the end that AlwaysKeepAlive still finds something?